AFQ/models/dti.py

```diff
@@ -6,19 +6,86 @@
 
 from dipy.core.geometry import vector_norm
 from dipy.reconst import dti
+from scipy.special import gamma
 
 import AFQ.utils.models as ut
 
 __all__ = ["fit_dti", "predict"]
 
 
-def _fit(gtab, data, mask=None):
-    dtimodel = dti.TensorModel(gtab)
+def noise_from_b0(data, gtab, bvals, mask=None, b0_threshold=50):
+    """
+    Coped from vistasoft's dtiComputeImageNoise
+    https://github.com/vistalab/vistasoft
+    And converted to python...
+
+    Parameters
+    ----------
+    data_files : str or list
+        Files containing DWI data. If this is a str, that's the full path to a
+        single file. If it's a list, each entry is a full path.
+    bval_files : str or list
+        Equivalent to `data_files`.
+    mask : ndarray, optional
+        Binary mask, set to True or 1 in voxels to be processed.
+        Default: Process all voxels.
+    b0_threshold : float
+    """
+    # make all-inclusive mask if None is provided
+    if mask is None:
+        mask = np.ones(data.shape[:3])
+
+    # Number of volumes in the dw dataset
+    num_vols = data.shape[3]
+
+    # Get brainmask indices
+    brain_inds = (mask > 0)
+
+    # preallocate a 2d array
+    # The first dimension is the number of volumes
+    # and the 2nd is each voxel (within the brain mask)
+    masked_data = np.zeros((num_vols, np.sum(brain_inds)))
+    # Loop over the volumes and assign the voxels within the brain mask
+    # to masked_data
+    for i in range(num_vols):
+        tmp = data[:, :, :, i]
+        masked_data[i, :] = tmp[brain_inds]
+
+    # Find which volumes are b=0
+    b0_inds = (bvals > b0_threshold)
+    n = len(b0_inds)
+    # Pull out the b=0 volumes
+    b0_data = masked_data[b0_inds, :]
+    # Calculate the median of the standard deviation. We do not think that
+    # this needs to be rescaled. Henkelman et al. (1985) suggest that this
+    # aproaches the true noise as the signal increases.
+    sigma = np.median(np.std(b0_data, axis=1, ddof=1))
+
+    # std of a sample underestimates sigma
+    # (see http://nbviewer.ipython.org/4287207/)
+    # This can be very big for small n (e.g., 20# for n=2)
+    # We can compute the underestimation bias:
+    bias = sigma * (1. - np.sqrt(2. / (n - 1))
+                    * (gamma(n / 2.) / gamma((n - 1) / 2.)))
+
+    # and correct for it:
+    return sigma + bias
+
+
+def _fit(gtab, data, mask=None, sigma=None):
+    if sigma is None:
+        dtimodel = dti.TensorModel(gtab)
+    else:
+        dtimodel = dti.TensorModel(
+            gtab,
+            fit_method="RT",
+            sigma=sigma)
     return dtimodel.fit(data, mask=mask)
 
 
 def fit_dti(data_files, bval_files, bvec_files, mask=None,
-            out_dir=None, file_prefix=None, b0_threshold=50):
+            out_dir=None, file_prefix=None, b0_threshold=50,
+            rtf=False):
```

Comment:
Looks like this key-word argument is not being used

Reply:
Also, can we use something more explicit? For example `restore=False` instead of `rtf`?

Reply:
To include restore in this function, we need to write another function that allows us to load multiple bval and bvec files then merge them together, I think. Which I have not done yet - the AFQ object does not use this function, but I think the examples do.

Reply:
I think `ut.prepare_data` does what you want? It's the first thing this function calls. 

In general, as you noted, this function is not directly called by the API. 

When we began the project, @jyeatman and folks in his team wanted to be able to fit DIPY models from the command line, which is why we still have a bunch of CLIs that do that. I wonder whether we should remove functionality directly related just to that. I believe that DIPY now has CLIs that do all these things. @jyeatman : WDYT?